rust/agama-network/src/nm/model.rs

```diff
@@ -0,0 +1,161 @@
+//! Set of structs and enums to handle devices and connections from NetworkManager.
+//!
+//! This are meant to be used internally, so we omit everything it is not useful for us.
+
+/// NetworkManager wireless mode
+///
+/// Using the newtype pattern around an String is enough. For proper support, we might replace this
+/// struct with an enum.
+use crate::{
+    model::{DeviceType, IpMethod, SecurityProtocol, WirelessMode},
+    nm::error::NmError,
+};
+use std::fmt;
+
+#[derive(Debug, PartialEq)]
+pub struct NmWirelessMode(pub String);
+
+impl Default for NmWirelessMode {
+    fn default() -> Self {
+        NmWirelessMode("infrastructure".to_string())
+    }
+}
+
+impl From<&str> for NmWirelessMode {
+    fn from(value: &str) -> Self {
+        Self(value.to_string())
+    }
+}
+
+impl NmWirelessMode {
+    pub fn as_str(&self) -> &str {
+        &self.0.as_str()
+    }
+}
+
+impl From<NmWirelessMode> for WirelessMode {
+    fn from(value: NmWirelessMode) -> Self {
+        match value.as_str() {
+            "infrastructure" => WirelessMode::Infra,
+            "adhoc" => WirelessMode::AdHoc,
+            "mesh" => WirelessMode::Mesh,
+            "ap" => WirelessMode::AP,
+            _ => WirelessMode::Other,
```

Comment:
hmm, so other acts like `unknown` here? I see same issue here. If we add another mode, we will need to renumber other or other won't be the last in enum. So again suggest to place it at first or somehow skip it.

Reply:
Yes, you are right. I will proceed in the same way that with the rest.